_posts/2015-08-18-How-katas-can-help-you-learn.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: post
+name: How-katas-can-help-you-learn
+title: How katas can help you learn
+date: 2015-08-18 09:35:00 +00:00
+author: Ana Nogal
+image:
+    src: /assets/img/custom/blog/baton_long.jpg
+    attribution:
+        text: Baton_long by Alain Delmas
+        href: https://commons.wikimedia.org/wiki/File:Baton_long.jpg
+tags:
+- Kata
+- TDD
+- iOS
+- Objective-C
+- Learning
+---
+In the last couple of weeks I have spent my time doing katas to improve my Objective-C. As you may remember, I even had an issue, [that you can revisit here](http://wp.me/p4i8Xl-7h).
+
+I've learn a lot from other katas too: for instance the RomanNumerals kata... You think it's a simple one, that you wan't learn mush from. But it's a good exercise  for your red -> green -> refactor cycle. It's a simple algorithm nothing new will come from here... Wasn't I wrong!!!!!
+
+I started  my kata, made tests then refactor. Add more tests and on the refactor phase, I decide to used an NSDictionary to map from a Decimal to a Roman number. Something like this:
+
+```
+NSDictionary *mapper = @{@10: @"X", @5:@"V", @1: @"I"};
+```
+
+That's when I discovered that the NSDictionary does not guarantee insertion order. What???? So my keys where all mixed up... And I needed them to maintain their order!!!! That's ok! I decided to create a class to map between decimals and romans. That's cool! Done! In the converter I just added a private property of type NSMutableArray. In it's constructor I added the class to the array all elements I needed. But now I have to do:
```

Comment:
were all mixed up
be consistent about capitalising Decimal and Roman numbers
In it's constructor I added the class to the array for all elements I needed.